metagraph-plugin/metagraph_katana/tests/test_algorithms.py

```diff
@@ -0,0 +1,247 @@
+    assert isinstance(cc_kg, dict)
+    assert isinstance(cc_kg, dict)
+    assert cc_kg == cc_nx
+
+
+# PageRank
+# mg.algos.centrality.pagerank(graph: Graph(edge_type=’map’, edge_dtype={‘int’, ‘float’}), damping: float = 0.85, maxiter: int = 50, tolerance: float = 1e-05) → NodeMap
+# the results of these 2 approaches are compared:
+# (1) katanagraph->nxgraph + nx's PageRank (2) nxgraph + nx's PageRank
+def test_pagerank(nx_weighted_directed_8_12, kg_from_nx_di_8_12):
+    pr_nx = mg.algos.centrality.pagerank(nx_weighted_directed_8_12)
+    pr_kg = mg.algos.centrality.pagerank(kg_from_nx_di_8_12)
+    assert isinstance(pr_nx, dict)
+    assert isinstance(pr_kg, dict)
+    assert pr_nx == pr_kg
+    
+
+# betweenness centrality
+# mg.algos.centrality.betweenness(graph: Graph(edge_type=’map’, edge_dtype={‘int’, ‘float’}), nodes: Optional[NodeSet] = None, normalize: bool = False) → NodeMap
+# the results of these 2 approaches are compared:
+# (1) katanagraph->nxgraph + nx's betweenness centrality (2) nxgraph + nx's betweenness centrality
+def test_betweenness_centrality(nx_weighted_directed_8_12, kg_from_nx_di_8_12):
+    bc_nx = mg.algos.centrality.betweenness(nx_weighted_directed_8_12)
+    bc_kg = mg.algos.centrality.betweenness(kg_from_nx_di_8_12)
+    assert isinstance(bc_nx, dict)
+    assert isinstance(bc_kg, dict)
+    assert bc_nx == bc_kg
+
+# triangle counting
+# mg.algos.clustering.triangle_count(graph: Graph(is_directed=False)) → int
+# the results of these 2 approaches are compared:
+# (1) katanagraph->nxgraph + nx's triangle counting (2) nxgraph + nx's triangle counting
+def test_triangle_counting(nx_weighted_undirected_8_12, kg_from_nx_ud_8_12):
+    tc_nx = mg.algos.clustering.triangle_count(nx_weighted_undirected_8_12)
+    tc_kg = mg.algos.clustering.triangle_count(kg_from_nx_ud_8_12)
+    assert isinstance(tc_nx, int)
+    assert isinstance(tc_kg, int)
+    assert tc_nx == tc_kg
+
+
+# Louvain community detection
+# mg.algos.clustering.louvain_community(graph: Graph(is_directed=False, edge_type=’map’, edge_dtype={‘int’, ‘float’})) → Tuple[NodeMap, float]
+# the results of these 2 approaches are compared:
+# (1) katanagraph->nxgraph + nx's Louvain community detection (2) nxgraph + nx's Louvain community detection
+def test_louvain_community_detection(nx_weighted_undirected_8_12, kg_from_nx_ud_8_12):
+    lc_nx = mg.algos.clustering.louvain_community(nx_weighted_undirected_8_12)
+    lc_kg = mg.algos.clustering.louvain_community(kg_from_nx_ud_8_12)
+    assert isinstance(lc_nx[0], dict)
+    assert isinstance(lc_kg[0], dict)
+    assert isinstance(lc_nx[1], float)
+    assert isinstance(lc_kg[1], float)
+    assert lc_nx[0] == lc_kg[0]
+    assert lc_nx[1] == lc_kg[1]
+
+
+# subgraph extraction
+# mg.algos.subgraph.extract_subgraph(graph: Graph, nodes: NodeSet) → Graph
+# the results of these 2 approaches are compared:
+# (1) katanagraph->nxgraph + nx's subgraph extraction (2) nxgraph + nx's subgraph extraction
```

Comment:
Use doc strings not comments above the functions.

Reply:
Okay, these are copied from metagraph's docstring. I think we can then remove them.